src/mlpack/core/cv/metrics/r2_score_impl.hpp

```diff
@@ -42,11 +43,20 @@ double R2Score::Evaluate(MLAlgorithm& model,
   // Calculate the denominator i.e.total sum of squares.
   double totalSumSquared = arma::accu(arma::square(responses - meanResponses));
 
-  // Handling undefined R2 Score when both denominator and numerator is 0.0.
-  if (residualSumSquared == 0.0)
-    return totalSumSquared ? 1.0 : DBL_MIN;
-
-  return 1 - residualSumSquared / totalSumSquared;
+  if (AdjustedR2)
+  {
+    // Handling undefined R2 Score when both denominator and numerator is 0.0.
+    if (residualSumSquared == 0.0)
+      return totalSumSquared ? 1.0 : DBL_MIN;
+    // Returning adjusted R-squared.
+    double rsq = 1 - (residualSumSquared / totalSumSquared);
+    return (1 - ((1 - rsq) * ((data.n_cols - 1) / (data.n_cols - data.n_rows - 1))));
```

Comment:
```suggestion
    return (1 - ((1 - rsq) * ((data.n_cols - 1) /
        (data.n_cols - data.n_rows - 1))));
```
This line was longer than 80 characters, so I wrapped it. :+1: